contracts/cw-proposal-multiple/src/contract.rs

```diff
@@ -646,9 +649,61 @@ pub fn query(deps: Deps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
         } => query_reverse_proposals(deps, env, start_before, limit),
         QueryMsg::ProposalHooks {} => to_binary(&PROPOSAL_HOOKS.query_hooks(deps)?),
         QueryMsg::VoteHooks {} => to_binary(&VOTE_HOOKS.query_hooks(deps)?),
+        QueryMsg::FilterListProposals {
+            wallet,
+            status,
+            wallet_vote,
+            start_after,
+            limit,
+        } => {
+            query_filter_list_proposals(deps, env, wallet, status, wallet_vote, start_after, limit)
+        }
     }
 }
 
+fn query_filter_list_proposals(
+    deps: Deps,
+    env: Env,
+    wallet: String,
+    status: Option<Status>,
+    wallet_vote: Option<MultipleChoiceVote>,
+    start_after: Option<u64>,
+    limit: Option<u64>,
+) -> StdResult<Binary> {
+    let wallet: Addr = deps.api.addr_validate(&wallet)?;
+    let min = start_after.map(Bound::exclusive);
+    let limit = limit.unwrap_or(DEFAULT_LIMIT);
+    let mut last_proposal_id = 0;
+
+    let props: Vec<ProposalResponse> = PROPOSALS
+        .range(deps.storage, min, None, cosmwasm_std::Order::Ascending)
+        .filter_map(|res| match res {
+            Ok((p_id, prop)) => {
+                last_proposal_id = p_id;
+                if status.map_or(true, |st| st == prop.status) {
+                    match BALLOTS.may_load(deps.storage, (p_id, wallet.clone())) {
+                        Ok(Some(ballot)) => {
+                            if wallet_vote.map_or(true, |v| v == ballot.vote) {
+                                return Some(prop.into_response(&env.block, p_id));
+                            }
+                        }
+                        Err(err) => return Some(Err(err)),
+                        _ => (),
+                    }
+                }
```

Comment:
If no wallet vote filter is applied, how would you feel about returning the proposal here if they have not voted?

This would let us find "open but not voted on" proposals by passing in `None` for wallet vote and `Open` for status. Something like this should do that (haven't actually tested this code):

```suggestion
                if status.map_or(true, |st| st == prop.status) {
                    if let Some(wallet_vote_filter) = wallet_vote {
                        if let Some(ballot) = BALLOTS.may_load(deps.storage, (p_id, wallet.clone()))? {
                            if ballot == wallet_vote_filter {
                                // Proposal has correct status and wallet vote.
                                return Some(prop.into_response(&env.block, p_id));
                            }
                        }
                    } else {
                         // No wallet vote filter specified so the proposal having the correct status is enough.
                         return Some(prop.into_response(&env.block, p_id));
                    }
                }
```

Reply:
Wow, I like that idea! But what if fhey want to find all the proposals they voted on,  with any vote?Imo good to have both options, maybe in different query. @ezekiiel Wdyt?

Reply:
Oh, great point. :) Maybe we could make the `wallet_vote` field an enum?

```rust
pub enum WalletVote {
  Voted {
    position: Option<SingleChoiceVote>,
  },
  DidNotVote {},
}
```

Reply:
Also for `DidNotVote` we want `voting_power` to be > 0, right?